### PR TITLE
Move PyInterpreter code in `python_variable.cpp` to its own files

### PR DESCRIPTION
**Pull request by kurtamohler**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #92648
* __->__ #92647

Part of #91395

cc @ezyang @bhosmer @smessmer @ljk53 @bdhirsh

### CODE REVIEW

**Review by ezyang — APPROVED** [review]
did only a cursory review, but long overdue move lol

File: torch/csrc/autograd/python_variable.cpp [hunk 1]
```diff
@@ -1,6 +1,4 @@
 #include <ATen/NamedTensorUtils.h>
-#include <ATen/core/PythonFallbackKernel.h>
```
Comment (albanD) [inline torch/csrc/autograd/python_variable.cpp:2]: Any more headers we can remove by any chance? :)

**Review by albanD — APPROVED** [review]
Thanks a lot!

File: torch/csrc/autograd/python_variable.cpp (hunk 1 above)
Comment (kurtamohler) [reply torch/csrc/autograd/python_variable.cpp:2]: Yeah looks like there are some. I'll follow up in another PR

Comment (albanD) [reply torch/csrc/autograd/python_variable.cpp:2]: Sure sounds good

**Review by albanD — APPROVED** [review]

**Pull request closed without merging**